public/controllers/agent/components/register-agent.js

```diff
@@ -716,7 +1344,7 @@ export const RegisterAgent = withErrorBoundary(
               <div>{guide}</div>
             ),
         },
-        ...(this.state.selectedOS == 'rpm' || this.state.selectedOS == 'deb'
+        ...(this.state.selectedOS == 'rpm' || this.state.selectedOS == 'cent' || this.state.selectedOS == 'suse' || this.state.selectedOS == 'fedora' || this.state.selectedOS == 'oraclelinux' || this.state.selectedOS == 'amazonlinux' || this.state.selectedOS == 'deb' || this.state.selectedOS == 'raspbian' || this.state.selectedOS == 'ubu' || this.state.selectedOS == 'win' || this.state.selectedOS == 'macos' || this.state.selectedOS == 'open' || this.state.selectedOS == 'sol' || this.state.selectedOS == 'aix' || this.state.selectedOS == 'hp'
```

Comment:
**nitpick**: I think it is painful and hard to maintain. We should look for another solution.

Reply:
I think the same as you, the development has been continued the way it was already, because the issue needed to be finished quickly.
The right thing to do would be to refactor the deploy new agent section, making clean, maintainable, scalable code.